js/views/message.js

```diff
@@ -70,10 +77,12 @@ views.Message = Backbone.Marionette.ItemView.extend({
 				},
 				type:'POST',
 				success: function () {
-					thisModel.get('flags').set('flagged', !starred);
+					// Retrieve the starred/flag status from the backend and display the results.
+					Mail.UI.messageView.loadMessages();
```

Comment:
same as above - no need to waste time on the network and create load on the server - we have all required information on the client